File: src/main/java/pl/khuzzuk/wfrp/helper/model/money/Price.java
package pl.khuzzuk.wfrp.helper.model.money;

import lombok.Data;
import pl.javahello.DTO;

import javax.persistence.Embeddable;

@DTO
@Data
@Embeddable
public class Price {
    private int gold;
    private int silver;
    private int lead;
}

File: src/wfrp-helper-web/src/data/world/money/Price.js
class Price {
    gold: number = 0;
    silver: number = 0;
    lead: number = 0;

    toString() {
        return this.gold + ' ' + this.silver + ' ' + this.lead;
    }

    updateWith(price: Price) {
        Object.assign(this, price);
    }
}

export default Price;